### PR TITLE
[rb] Run remote tests on RBE

### PR DESCRIPTION
**Pull request by p0deje**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [ ] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Excluded remote drivers from specific tests to prevent concurrent execution issues.
- Added RBE environment detection and adjusted test environment configurations accordingly.
- Updated CI workflow to refine test execution matrix and flags.
- Modified test configurations to remove unnecessary NO_SANDBOX setting for non-Windows platforms.
- Updated the list of skipped tests to include additional remote and BiDi tests.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>driver_spec.rb</strong><dd><code>Exclude remote drivers from concurrent start test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
rb/spec/integration/selenium/webdriver/driver_spec.rb

- Added exclusion for remote drivers in concurrent start test.

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13906/files#diff-f99e2caaa528d675ddc6279b1f1a74dde50ea5870a14dfffcbd45e6a03140693">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>profile_spec.rb</strong><dd><code>Exclude remote drivers from profile reuse test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
rb/spec/integration/selenium/webdriver/firefox/profile_spec.rb

- Added exclusion for remote drivers in profile reuse test.

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13906/files#diff-be2858754037bb63f1b861c64758de7aea64e6910f0449393deaa10fb02ae128">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>.skipped-tests</strong><dd><code>Update skipped tests list for remote and BiDi</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.skipped-tests

- Added multiple remote and BiDi tests to skipped tests list.

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13906/files#diff-f5b9e26559bdc4671c8a6fd4013e376a76cbdac0a1661ffbe585b7533c4700f7">+18/-5</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>spec_helper.rb</strong><dd><code>Add RBE environment condition</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
rb/spec/integration/selenium/webdriver/spec_helper.rb

- Added condition for RBE environment.

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13906/files#diff-9147927bbff15c8b0da05a9b44f258feaab8ef8bf1bf667de0f5fcd4eefea284">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test_environment.rb</strong><dd><code>Enhance test environment for RBE and driver configurations</code></dd></summary>
<hr>
      
rb/spec/integration/selenium/webdriver/spec_support/test_environment.rb

<li>Added RBE detection method.<br> <li> Modified remote server arguments based on environment variables.<br> <li> Included RBE status in current environment details.<br> <li> Adjusted Chrome and Edge options for non-Windows platforms.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13906/files#diff-884d159fa58dcbf2cdfec0003a9a72bf2a95fa661f3d2787ec801a3669521b6a">+19/-4</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>tests.bzl</strong><dd><code>Update test configurations for remote execution</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
rb/spec/tests.bzl

<li>Removed NO_SANDBOX setting for non-Windows platforms.<br> <li> Adjusted tags for remote tests.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13906/files#diff-0b035bed60fc62ad1db7c3548fa4df1aa1cd0ebe7b3c50cf5643a34929d8cb68">+1/-10</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ci-ruby.yml</strong><dd><code>Update CI workflow for Ruby tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.github/workflows/ci-ruby.yml

<li>Updated matrix configurations for local and remote tests.<br> <li> Adjusted pin_browsers flag usage.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13906/files#diff-e71b669ae33da2f4e6f9e0858c13b326eaea0388934d0a21e3ebc31005b7aa90">+6/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: rb/.ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-3.0.6
+jruby-9.4.5.0
```
Comment (titusfortner) [inline rb/.ruby-version]: is there a way to set this just for RBE? after working with both for this release, using 3.0.6 ended up being much easier.

Comment (titusfortner) [reply rb/.ruby-version]: I think I was running into problems using jruby with RubyMine, though I don't remember exactly what my problem was now, I kept doing and undoing things so often. :)

Comment (p0deje) [reply rb/.ruby-version]: No, I've reverted this.

File: .github/workflows/ci-ruby.yml [hunk 2]
```diff
@@ -74,14 +74,13 @@ jobs:
...
         os:
-          - ubuntu
           - windows
           - macos
         exclude:
-          - browser: edge
-            os: ubuntu
           - browser: edge
             os: macos
+          - browser: safari
```
Comment (titusfortner) [inline .github/workflows/ci-ruby.yml]: we pulled safari entirely for now because it wasn't running.

**Review by titusfortner — APPROVED** [review]
Looks great. Theoretically should we run all the RBE skipped tests on GitHub ubuntu?

**Pull request merged**